src/SAWScript/Crucible/LLVM/Override.hs

```diff
@@ -1608,28 +1608,37 @@ matchPointsToBitfieldValue opts sc cc spec prepost loc ptr bfIndex val =
                        -- Let us imagine that we are matching against the
                        -- `y` field. The bitfield (bfBV) will look something
                        -- like 0b????Y???, where `Y` denotes the value of the
-                       -- `y` bit.
-                       let -- The offset (in bits) of the field within the
-                           -- bitfield. For `y`, this is 3 (x1's offset is 0
-                           -- and `x2`'s offset is 1).
-                           bfOffset = biFieldOffset bfIndex
-                           bfOffsetBV = BV.mkBV bfWidth $ fromIntegral bfOffset
-
-                       -- Use a logical right shift to make the bits of the
-                       -- field within the bitfield become the least
-                       -- significant bits. In the `y` example, this is
-                       -- tantamount to shifting 0b????Y??? right by `bfOffset`
-                       -- (i.e., 3 bits) to become 0b000????Y.
-                       bfBV'  <- liftIO $ W4.bvLshr sym bfBV =<< W4.bvLit sym bfWidth bfOffsetBV
-
-                       -- Finally, truncate the bitfield such that only the
-                       -- bits for the field remain. In the `y` example, this
-                       -- means truncating 0b000????Y to `0bY`.
-                       bfBV'' <- liftIO $ W4.bvTrunc sym rhsWidth bfBV'
-
-                       -- Match the truncated bitfield against the RHS value.
-                       let res_val' = Crucible.LLVMValInt bfBlk bfBV''
-                       pure Nothing <* matchArg opts sc cc spec prepost res_val' memTy val
+                       -- `y` bit. Here, `bfWidth` (the width of the entire
+                       -- bitfield in bits) is 8, and `rhsWidth` (the width of
+                       -- the `y` field in bits) is 1.
+
+                       -- The offset (in bits) of the field within the
+                       -- bitfield. For `y`, this is 3 (x1's offset is 0 and
+                       -- `x2`'s offset is 1).
+                       Some bfOffset <- pure $ mkNatRepr $ fromIntegral
+                                             $ biFieldOffset bfIndex
+
+                       -- Next, convince the typechecker that
+                       -- (bfOffset + rhsWidth) <= bfWidth. This should always
+                       -- be the case if the BitfieldIndex is constructed
+                       -- correctly.
+                       LeqProof <- case testLeq (addNat bfOffset rhsWidth) bfWidth of
+                         Just prf -> pure prf
+                         Nothing -> fail $ unlines
```

Comment:
I think this is worth upgrading to a `panic`.  Let's try to avoid fresh uses of `fail` whenever we can.

Reply:
Oops, I forgot that `panic` was a thing. I've replaced this `fail` with a `panic`.